### PR TITLE
Update to node 12 and fix dep issues

### PR DESCRIPTION
**Pull request by samccone**

### CODE REVIEW

**Comment by samccone** [comment]
cc @peterblazejewicz @jdaudier

File: gruntfile.js [hunk 1]
```diff
@@ -80,7 +80,6 @@ module.exports = function(grunt) {
         compress: true,
         mangle: false,
         preserveComments: false,
-        screwIE8: true,
```
Comment (samccone) [inline gruntfile.js]: https://github.com/gruntjs/grunt-contrib-uglify/issues/330

File: tasks/utils/gitty-cache.js [hunk 2]
```diff
@@ -13,12 +13,15 @@ module.exports = {
           return this.sortedTags;
         }
 
-        return repo.tagsAsync()
```
Comment (samccone) [inline tasks/utils/gitty-cache.js]: https://github.com/gordonwritescode/gitty/blob/92bfeabc30addab9a7e362a83c670a25b022f33c/lib/repository.js#L425

**Comment by jasonLaster** [comment]
:100:

**Comment by trezy** [comment]
:+1: Works well. Fixed all my woes. :-D

**Comment by peterblazejewicz** [comment]
The PR fixed this `NPM install` error for me:
```
pty.js@0.2.4 install /Users/piotrblazejewicz/git/marionettejs.com/node_modules/gitty/node_modules/pty.js
> node-gyp rebuild
...
14 errors generated.
make: *** [Release/obj.target/pty/src/unix/pty.o] Error 1
gyp ERR! build error 
...
```

**Comment by jdaudier** [comment]
i :heart:  the `screwIE8` config

**Comment by samccone** [comment]
What version of 12

**Comment by peterblazejewicz** [comment]
> What version of 12

If it's to me, no, this was under Node 4.2.1

**Comment by samccone** [comment]
Oh yeah only works for 12 :(

**Comment by trezy** [comment]
We should probably add the `engines` config to the `package.json`, as well as a note in the README just to make it clear that it only works with 0.12.x.

**Comment by samccone** [comment]
I just tested a clean install on node 4 and it all seems to work (for me)

**Pull request merged**